File: GO/euler003.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(solve())
}

func solve() int {
	LIMIT := 600851475143

	for i := int(math.Sqrt(float64(LIMIT))); i > 1; i-- {
		//fmt.Println(i)
		if LIMIT%i == 0 && is_prime(i) {
			return i
		}
	}
	return -1
}

func is_prime(p int) bool {
	for i := 2; i <= int(math.Sqrt(float64(p))); i++ {
		if p%i == 0 {
			return false
		}
	}
	return true
}

File: C#/euler003.cs
using System;

namespace ProjectEuler {
    class euler003 {
        static void Main() {
            Console.WriteLine(Solve());
        }

        static int Solve() {
            double num = 600851475143;
            int end = Convert.ToInt32(Math.Sqrt(num));

            for (int i = end; i > 1; i--) {
                if (is_prime(i) && num%i == 0) {
                    return i;
                }
            }
            return -1;
        }

        static bool is_prime(int n) {
            int end = Convert.ToInt32(Math.Sqrt(n));
            for (int i = 2; i <= end; i++) {
                if (n%i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}